CMSgov/mdct-mcr
# MDCT-1960 pt. 3 - Lockdown archived

## Description
[MDCT-1960](https://qmacbis.atlassian.net/browse/MDCT-1960) 
1. Make it so state users cannot manually switch to an archived report
2. Return a 403 if they attempt to edit an archived report (best tested in Postman).

### How to test
To test case 1
1. Login as state user and make 2 reports. Enter one of them and fill in a page.
3. Login as an admin and archive the report you entered data into.
4. Login as a state user and open the other report
5. From dynamodb-admin get the id of the archived report
6. Open Inspect -> Application -> Local Storage and change the `selectedReportId` to the archived report
7. Refresh the page, it should kick you to the dashboard.

(you can test the opposite case, where it doesn't kick you and you successfully see the other report, on main or by removing the ui-src code in this PR)

To test case 2
Create a report and archive it
Hit it with an update from postman ([collection here](https://qmacbis.atlassian.net/l/cp/r8FNquPT)) and see the 403
Unarchive and see update works (there's a nice postman call for archive/unarchive 😉 )

## Code author checklist

## Reviewer checklist (two different people)
